esy-install/DistStorage.re

```diff
@@ -67,15 +67,52 @@ let ofCachedTarball = path =>
   Tarball({tarballPath: path, stripComponents: 0});
 let ofDir = path => SourcePath(path);
 
-let fetch' = (sandbox, dist, gitUsername, gitPassword) => {
+let fetch' = (sandbox, dist, pkg, gitUsername, gitPassword) => {
```

Comment:
Is it possible to just make do with `extraSources` field instead of `Package.t`?

Reply:
I made `extraSources` optional argument instead of passing the whole `Package.t`